src/ReplicatedStorage/Client/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,63 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local ContextActionService = game:GetService("ContextActionService")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local clientComm = Comm.ClientComm.new(character, false, "Sprint")
```

Comment:
Try not to construct things above self - it makes the function half virtual half method

Reply:
This was done because ClientComm is a yield function. Probably makes more sense for me to move this down though.